### PR TITLE
options/posix: implement getservbyport()

### LINKED ISSUE #289: getservbyport() unimplemented
The following program, tested on managarm, require `getservbyport()` to properly function

- `nc`, without it, `nc` has no working features.

### PR DESCRIPTION
**Pull request by Geertiebear**
Fixes #289.

### CODE REVIEW

File: options/posix/generic/services.cpp [hunk 1]
```diff
@@ -11,7 +11,88 @@
...
+			int rport = strtoul(pos, &end, 10);
+			if (rport != port || rport > 65535) {
+				pos = end;
+				continue;
+			}
+
+			// We have found the port, time to rewind to the start
+			// of the line.
+			for (; pos[-1]; pos--)
+				name_length++;
```
Comment (qookei) [inline options/posix/generic/services.cpp:70]: Is the character before the line buffer guaranteed to be 0?

Comment (Geertiebear) [reply options/posix/generic/services.cpp:70]: Good point, it actually isn't, but I can fix that real quick. Is the rest okay?

Comment (qookei) [reply options/posix/generic/services.cpp:70]: The `name_length` calculation seems like it includes the whitespace between the name and port as well, is that expected?

Comment (Geertiebear) [reply options/posix/generic/services.cpp:70]: Hm that's true, I guess that isn't expected. I'll add a check for whitespace too.

**Review by qookei — APPROVED** [review]
LGTM

**Pull request merged**